pkg/webhook/cluster/nutanix_validator.go

```diff
@@ -125,3 +134,33 @@ func checkIfPrismCentralIPInLoadBalancerIPRange(
 
 	return nil
 }
+
+// checkIfPrismCentralAndControlPlaneIPSame checks if Prism Central and Control Plane IP are same.
+// It compares strictly IP addresses(no FQDN) and doesn't involve any network calls.
+// This is a temporary check until we have a better way to handle this by reserving IPs
+// using IPAM provider.
+func checkIfPrismCentralAndControlPlaneIPSame(
+	pcEndpoint v1alpha1.NutanixPrismCentralEndpointSpec,
+	controlPlaneEndpointSpec v1alpha1.ControlPlaneEndpointSpec,
+) error {
+	controlPlaneEndpointIP, err := netip.ParseAddr(controlPlaneEndpointSpec.VirtualIPAddress())
+	if err != nil {
+		// If controlPlaneEndpointIP is a hostname, we cannot compare it with PC IP
+		// so return directly.
+		return nil
+	}
+
+	pcHostname, _, err := pcEndpoint.ParseURL()
+	if err != nil {
+		return err
+	}
+
+	pcIP, err := netip.ParseAddr(pcHostname)
```

Comment:
This looks very similar to https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/1001/files#diff-cbaf86208d1803beb69b004fc24fe8d3dccbc2cba74f10dde9f8e8acf5edebeeR98-R107 but is using slightly different logic, can we reuse some util function please?

Reply:
21293e9ffa7fd596f3ba894845b1c7f3f74f177c